### DISCUSSION TITLE
what tooling/ide you are using to debug runtime it in Linux?

### DISCUSSION THREAD

**Discussion by minxuanz** [dotnet/runtime#94199] [Q&A]
I'm new to Linux and I'm having some trouble with vim and gdb.

**Comment by gfoidl** [comment]
Do you mean something like [Debugging CoreFX on Unix](https://github.com/dotnet/runtime/blob/main/docs/workflow/debugging/libraries/unix-instructions.md#debugging-corefx-on-unix)?
In short it's LLDB with SOS-extensions.

Or what do you mean by "debug runtime" exactly? Any .NET program that runs or the basic .NET runtime?

**Reply by minxuanz** [reply]
[debugging-runtime.md](https://github.com/dotnet/runtime/blob/main/docs/workflow/debugging/coreclr/debugging-runtime.md#debugging-coreclr)，I want to find  a ide to debugging CoreCLR on Linux,like Visual Studio  on windows. 
Thanks for your reply.

**Reply by gfoidl** [reply]
AFAIK there's no IDE (like VS) for that. The CLI is your friend.
Maybe you find an IDE for LLDB, but usually you are way faster and more productive by using the CLI.

**Reply by MichalPetryka** [reply]
> AFAIK there's no IDE (like VS) for that. The CLI is your friend.
> Maybe you find an IDE for LLDB, but usually you are way faster and more productive by using the CLI.

Could CLion with LLDB set as the debugger maybe work?
